web/packages/teleport/src/services/auth/auth.ts

```diff
@@ -442,27 +434,33 @@ const auth = {
       .then(res => res?.webauthn_response);
   },
 
-  getMfaChallengeResponseForAdminAction(allowReuse?: boolean) {
-    // If the client is checking if MFA is required for an admin action,
-    // but we know admin action MFA is not enforced, return early.
-    if (!cfg.isAdminActionMfaEnforced()) {
-      return;
+  getAdminActionMfaResponse(allowReuse?: boolean) {
+    // mfaContext is set once the react-scoped MFA Context Provider is initialized.
+    // Since this is a global object outside of the react scope, there is a marginal
+    // chance for a race condition here (the react scope should generally be initialized
+    // before this has a chance of being called). This conditional is not expected to
+    // be hit, but will catch any major issues that could arise from this solution.
+    if (!mfaContext) {
+      setTimeout(() => {
+        if (!mfaContext)
+          throw new Error(
+            'Failed to set up MFA prompt for admin action. This is a bug.'
```

Comment:
Does this surface to the user? Should we be more helpful than "This is a bug"?

Reply:
Hm, looking at this again - this error doesn't go anywhere, as it's thrown within a `setTimeout`

Reply:
You're right, for now I'll remove the timeout and just return a better error message:

`Failed to set up MFA prompt for admin action. Please try refreshing the page to try again. If the issue persists, contact support as this is likely a bug.`